pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -193,6 +193,10 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 	processNode := func(i int) {
 		nodeInfo := nodeNameToInfo[allNodeNames[i]]
 		if nodeInfo.Node() != nil {
+			//we just need to calculate the filteredNodes 
+			if !nodeInfilteredNodes(nodeInfo, nodes) {
```

Comment:
No, we need to check all nodes, as those may have running pods that have affinity/anti-affinity rules.

Reply:
The `nodes` here are "filteredNodes". I suppose "fiteredNodes" already satisfy node/pod {anti-}affinity, so IMO they can be used safely for interpod priority check.

Or @bsalamat are you trying to say: interpod priority function is intended to **iterate pods on all nodes** to get a more reasonable score?

(but this fix isn't performant, I proposed a solution in https://github.com/kubernetes/kubernetes/pull/71133/files#r235215190)